### ISSUE TITLE
autodoc typehints description broken with commas

### ISSUE THREAD

**Issue by fcolas** [sphinx-doc/sphinx#7649]
**Describe the bug**
When using type hints containing a comma with the `'description'` option, the output is wrong.
For instance with:
```python
def search(path: Union[str, os.PathLike[str]])
    """Search."""
```
The output shows:
```
Parameters: os.PathLike[str]] path (Union[str,) –
```
instead of:
```
Parameters: path (Union[str, os.PathLike[str]]) –
```
It is the same with `Dict`, `Callable`, and most probably others like `Tuple`.

**Environment info**
- OS: macos 10.14.6
- Python version: 3.7.7
- Sphinx version: 3.0.3
- Sphinx extensions: `extensions = ['sphinx.ext.autodoc']`

**Comment by tk0miya** [comment]
Thank you for reporting. This was fixed at #7562.
Please wait for a next release.

**Comment by fcolas** [comment]
Sorry for the duplicate and thanks for your work.
